### PR TITLE
CSS: Add a reset for image heights

### PR DESCRIPTION
**Pull request by scruffian**
## Description (updated 5/10/2021)
For all images that explicitly set widths or max-widths of 100% we should add a `height: auto` rule to prevent them from becoming stretched.

## Screenshots 
Before:
localhost_4759__p=435

After:
localhost_4759__p=435 (1)

## How has this been tested?
- Switch to empty theme
- Add an image to a post
- Check that the image isn't stretched in the front end
- Try setting wide and full alignments
- Also try a media and text block

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/common.scss [hunk 1]
```diff
@@ -81,3 +81,7 @@
 .items-justified-space-between {
 	justify-content: space-between;
 }
+
+img {
+	height: auto;
```
Comment (youknowriad) [inline packages/block-library/src/common.scss]: Not sure I understand the issue and fix here. I tried using an image and it didn't seem stretched for me. Maybe there are special cases? Anyway, it seems this is overriding something that comes from Core itself. Is that style broken? (inline image height) Should we remove it in Core itself directly. 

cc @felixarntz as it seems related to your work on lazy loading images.

Comment (scruffian) [reply packages/block-library/src/common.scss]: It depends on the proportions of your image. You can see it in https://maylandblockstarter.wordpress.com/

Comment (youknowriad) [reply packages/block-library/src/common.scss]: This is indeed not great, it seems we shouldn't be including that height in the first place, shouldn't we?

Comment (kjellr) [reply packages/block-library/src/common.scss]: I can reproduce this for image blocks when testing the [`emptytheme`](https://github.com/WordPress/theme-experiments/tree/1a4128f7c2c741e9c14b4c2a6bffd9c5b06e77e2/emptytheme) when:

- There's a wide or full alignment selected.
- The image has "Full Size" set as [the image size](https://cloudup.com/cQIEXGcPyG6). 

When an image block is inserted, it gets some height and width by default. But when Gutenberg adjusts those, things get stretched. I'm not sure whether the height should be fixed by Gutenberg or by core then, but it probably shouldn't need to be fixed in the themes. 🙂 

---

**Default, Large Size** (Things look fine in this scenario)

Editor|Front
---|--- 
Screen Shot 2021-03-22 at 12 08 17 PM|Screen Shot 2021-03-22 at 12 11 04 PM

**Wide, Large Size**

Editor|Front
---|--- 
Screen Shot 2021-03-22 at 12 09 10 PM|Screen Shot 2021-03-22 at 12 09 18 PM

**Default, Full Size**

Editor|Front
---|---
Screen Shot 2021-03-22 at 12 14 50 PM|Screen Shot 2021-03-22 at 12 15 04 PM

Comment (youknowriad) [reply packages/block-library/src/common.scss]: > but it probably shouldn't need to be fixed in the themes

I agree with this yes

**Comment by jasmussen** [comment]
Simple fix that seems to do the job, and could potentially pass as a bandaid. But isn't this an issue that should be fixed at the source?

**Comment by youknowriad** [comment]
yes, this basically undo what the core ticket does a little bit, so I'd like to have @felixarntz's thoughts on the best way forward here. I'm thinking personally that we should revert the addition of these inline width/heights because they don't take into consideration the context of where the image is inserted.

File: packages/block-library/src/common.scss (hunk 1 above)
Comment (felixarntz) [reply packages/block-library/src/common.scss]: In principle, IMO this is the responsibility of themes to provide. Themes supporting the block editor have the responsibility of providing styles for `.alignwide` and `.alignfull`. As part of that (and as part of any usage that makes images span a certain width), it's necessary that any image has its height reset. So I'd argue, every theme should e.g. include something like
```css
img.alignwide,
img.alignfull {
    height: auto;
}
```

I agree though that it makes sense to solve this in Gutenberg because a theme developer may not be aware of WordPress images potentially having size attributes, and it is a trivial fix. We're certainly solving this widely by including it in Gutenberg itself.

**Comment by felixarntz** [comment]
@youknowriad I left some feedback in https://github.com/WordPress/gutenberg/pull/30092/files#r599874756.

TLDR I _do_ think it is a bandaid of somewhat broken theme styles, _but_ it should be a reliable and trivial bandaid, so I'm not opposed to having it included in Gutenberg/WordPress.

The case against it would be: Before there was Gutenberg, almost all images in WordPress already had size attributes on them. So if a theme had something like `img { max-width: 100%: }`, it would have already needed the reset `img { height: auto; }` to not run into the same problems that is surfaced here. Potentially this is a matter of clarifying in the documentation.

But Gutenberg _does_ include that styling itself for the image block, so I think it's worth including it there specifically.

**Review by felixarntz — COMMENTED** [review]
Instead of resetting the height for _any_ images (which should remain responsibility of themes), I suggest to only reset height for images where Gutenberg itself defines the styles that would break it without reset, which is in https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/image/style.scss#L6.

File: packages/block-library/src/common.scss (hunk 1 above)
Comment (youknowriad) [reply packages/block-library/src/common.scss]: > In principle, IMO this is the responsibility of themes to provide.

This goest against the principles of Full Site Editing. In Full Site Editing, the theme doesn't have to provide anything for things to work properly and to have the editor match the frontend. You can start with an empty theme, go to the site editor and start building templates without having to provide any custom CSS.

Based on this principle, I don't think it makes sense anymore to have these inline attributes (the height at least) anymore in WordPress. There are just too many situations where things could go wrong. Thing just random images inside columns, inside a small "group" block. Images widths are not going to match the actual image width in a lot of situations, responsive mode is another one too.

So I'm not sure it makes sense to add these inline attributes if we override the styles later in a stylesheet.

Comment (scruffian) [reply packages/block-library/src/common.scss]: >  Themes supporting the block editor have the responsibility of providing styles for .alignwide and .alignfull. 

This is no longer the case for themes that opt into the new alignments offered in FSE (https://github.com/WordPress/gutenberg/pull/29335)

**Comment by ellatrix** [comment]
@jasmussen This is also an issue in TT1 Blocks that I have also noticed. https://github.com/WordPress/theme-experiments/issues/259

Strangely there, it only happens for the Media + Text block.

**Comment by youknowriad** [comment]
Circling back to this issue. After some reading on the net, it seems we should do it. Basically all images should have height and width HTML attributes defined and wherever there's `width: 100%` or `max-width: 100%` in CSS for images, add the height auto rule as well.

So what do you think about moving this rule to these places where we have the 100% widths instead of doing a global rule like that?

**Comment by kjellr** [comment]
> wherever there's width: 100% or max-width: 100% in CSS for images, add the height auto rule as well.

That seems reasonable to me. 👍

**Comment by scruffian** [comment]
> So what do you think about moving this rule to these places where we have the 100% widths instead of doing a global rule like that?

I've done this in 55ad9e2, so this could use another review. Sorry it took so long!

**Review by MaggieCabrera — APPROVED** [review]
I say yes to this, please

**Comment by youknowriad** [comment]
I didn't test but the changes look good to me

**Review by aristath — APPROVED** [review]

**Review by felixarntz — APPROVED** [review]

**Comment by scruffian** [comment]
Any idea how to get the performance tests to pass?

**Comment by youknowriad** [comment]
@scruffian did you try rebasing, these tests can fail sometimes with old code.
